content/middlewares.md

```diff
@@ -241,3 +241,5 @@ const app = await NestFactory.create(AppModule);
 app.use(logger);
 await app.listen(3000);
 ```
+
+> info **Hint** Accessing the DI container in a global middleware is not possible. You can use a [functional middleware](middleware#functional-middleware) instead when using `app.use()`. Alternatively, you can use a class middleware and consume it with `.forRoutes('*')`.
```

Comment:
```suggestion
> info **Hint** Accessing the DI container in a global middleware is not possible. You can use a [functional middleware](middleware#functional-middleware) instead when using `app.use()`. Alternatively, you can use a class middleware and consume it with `.forRoutes('*')` within any imported Nest module.
```
Might be a valuable addition(? not sure)

Reply:
Mhh I would cross out the `imported`. So just the `within any Nest module`. However, maybe it makes sense to put it this way:

```
and consume it with `.forRoutes('*')` within the AppModule (or any other module).
```

Reasoning behind this: In this case it's meant as an alternative to a global functional middleware in `app.use()`, so you would, for this use case, put it into the AppModule.

wdyt?

Or just create a full example, e.g.:

```typescript
import { Module, NestModule, MiddlewareConsumer } from '@nestjs/common';
import { LoggerMiddleware } from './common/middleware/logger.middleware';
import { CatsModule } from './cats/cats.module';

@Module({
  imports: [CatsModule],
})
export class AppModule implements NestModule {
  configure(consumer: MiddlewareConsumer) {
    consumer
      .apply(LoggerMiddleware)
      .forRoutes('*');
  }
}
```

Reply:
I like the suggestion with the `AppModule` addition (not the full code example as it's already available if you scroll up either way)

Reply:
Just updated my suggestion

Reply:
👍 I accepted the suggestion.